Migrate netapi tests to PyTest

s0undt3ch opened the pull request:
### What does this PR do?
See title
This is also the precursor for the fix to #56495 on https://github.com/saltstack/salt/pull/60493

s0undt3ch committed:
Add PAM ``eauth`` tests for ``rest_tornado``

s0undt3ch committed:
Support using ``Webserver`` as a context manager

s0undt3ch committed:
Fix ``tests/pytests/integration/netapi/rest_tornado/test_minions_api_handler.py`` tests

s0undt3ch committed:
Set the account password, even if reusing existing temporary accounts.

s0undt3ch committed:
Delete the home directory AFTER deleting the account on macOS

Ch3LL commented on tests/pytests/integration/netapi/test_ssh_client.py:
How about we use https://pytest-httpserver.readthedocs.io/en/latest/ here or something similar, so we are not depending on external resources.

s0undt3ch replied:
Yep, that might be a good approach going forward.

Ch3LL replied:
Did you want to handle that in this PR or a follow up one? I'm fine either way.

s0undt3ch replied:
This one still needs work, i have to revisit this.
But probably on a separate PR, where we address all webservers we use ...

s0undt3ch committed:
Migrate ``rest_tornado`` tests to PyTest(those possible)

s0undt3ch committed:
Renamed ``tests/support/saltnado.py`` -> ``tests/support/netapi.py``

s0undt3ch committed:
Migrate ``tests/integration/netapi/test_client.py`` to Pytest

s0undt3ch committed:
Remove ``external_auth`` from the static config file

s0undt3ch committed:
Migrate ``rest_cherrypy`` tests to pytest

s0undt3ch committed:
Fix wildcard eauth.

Migrate ``tests/integration/client/test_runner.py`` and ``tests/integration/wheel/`` to PyTest

Ch3LL commented:
Are there still plans to try to get this in for phosphorus or can I take off the phosphorus label?

s0undt3ch commented:
Switched to 3006 and converted to draft

s0undt3ch commented:
re-run full all
